src/lib/services/invoice-account-addresses-service.js

```diff
@@ -0,0 +1,33 @@
+const InvoiceAccountAddress = require('../models/invoice-account-address');
+const invoiceAccountsConnector = require('../connectors/crm-v2/invoice-accounts');
+const DateRange = require('../models/date-range');
+
+const getInvoiceAccountAddressModel = (startDate, addressModel, agentCompanyModel, contactModel) => {
+  const invoiceAccountAddress = new InvoiceAccountAddress();
+  return invoiceAccountAddress.fromHash({
+    dateRange: new DateRange(startDate, null),
+    address: addressModel,
+    agentCompany: agentCompanyModel,
+    contact: contactModel
+  });
+};
+
+const createInvoiceAccountAddress = async (invoiceAccount, startDate) => {
+  const { address, agentCompany, contact } = invoiceAccount.invoiceAccountAddresses[0];
+  const data = {
+    addressId: address.id,
+    startDate,
+    agentCompanyId: agentCompany ? agentCompany.id : null,
+    contactId: contact ? contact.id : null
+  };
+
+  const invoiceAccountAddress = await invoiceAccountsConnector.createInvoiceAccountAddress(invoiceAccount.id, data);
+  invoiceAccount.invoiceAccountAddresses[0].id = invoiceAccountAddress.invoiceAccountAddressId;
```

Comment:
Although I have done this elsewhere myself, I think it would be better not to manipulate the arguments and return a completely fresh model, e.g.:
```
  const invoiceAccountAddress = await invoiceAccountsConnector.createInvoiceAccountAddress(invoiceAccount.id, data);
return mappers.crm.invoiceAccountAddress.crmToModel(invoiceAccountAddress);

```